lib/bridge/instance.js

```diff
@@ -0,0 +1,175 @@
+/*eslint no-invalid-this: 0*/ // eslint doesn't understand Promise.coroutine wrapping
+"use strict";
+var Promise = require("bluebird");
+var extend = require("extend");
+
+var store = require("../store");
+var ircToMatrix = require("./irc-to-matrix.js");
+var matrixToIrc = require("./matrix-to-irc.js");
+var membershiplists = require("./membershiplists.js");
+var ircLib = require("../irclib/irc.js");
+var names = require("../irclib/names.js");
+var IrcServer = require("../irclib/server.js").IrcServer;
+var matrixLib = require("../mxlib/matrix");
+var MatrixUser = require("../models/users").MatrixUser;
+var logging = require("../logging");
+var log = logging.get("IrcBridge");
+var Bridge = require("matrix-appservice-bridge").Bridge;
+
+const DELAY_TIME_MS = 10 * 1000;
+const DEAD_TIME_MS = 60 * 1000 * 5; // 5min
+
+
+function IrcBridge(config, registration) {
+    this.config = config;
+    this.registration = registration;
+    this.ircServers = [];
+    this.bridge = null; // Bridge
+}
+
+IrcBridge.prototype.run = Promise.coroutine(function*(port) {
+    // connect to the DB, blow away any old config mappings, we're setting new ones now.
+    yield store.connectToDatabase(this.config.ircService.databaseUri);
```

Comment:
This is surprisingly side-effecty. Surely this should return a database connection, rather than presumably set global state?